src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl

```diff
@@ -125,9 +125,9 @@ fi
         echo
         echo Running pip install...
         START_TIME=$SECONDS
-        InstallCommand="$python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r $REQUIREMENTS_TXT_FILE --target="{{ PackagesDirectory }}" --upgrade | ts $TS_FMT"
+        InstallCommand="$python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r $REQUIREMENTS_TXT_FILE --target="{{ PackagesDirectory }}" {{ oryxPipUpgradeCommand }} | ts $TS_FMT"
         printf %s " , $InstallCommand" >> "$COMMAND_MANIFEST_FILE"
-        output=$( ( $python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r $REQUIREMENTS_TXT_FILE --target="{{ PackagesDirectory }}" --upgrade | ts $TS_FMT; exit ${PIPESTATUS[0]} ) 2>&1; exit ${PIPESTATUS[0]} )
+        output=$( ( $python -m pip install --cache-dir $PIP_CACHE_DIR --prefer-binary -r $REQUIREMENTS_TXT_FILE --target="{{ PackagesDirectory }}" {{ oryxPipUpgradeCommand }} | ts $TS_FMT; exit ${PIPESTATUS[0]} ) 2>&1; exit ${PIPESTATUS[0]} )
         pipInstallExitCode=${PIPESTATUS[0]}
 
         ELAPSED_TIME=$(($SECONDS - $START_TIME))
```

Comment:
nit: there are also references to the `--upgrade` flag in `pip install` commands down below on lines 146 and 148 -- do we need to use this same logic there?

Reply:
sorry Cormac, I have missed this comment to reply earlier.
We do not need to use the same logic here. The reason we added the logic is because `pip` does not go well along with `target` and `upgrade` together.